torch/fx/passes/tools_common.py

```diff
@@ -123,12 +123,19 @@ def recursive_add_node(
         self,
         fusion_group: "FxNetAccFusionsFinder.FusionGroup",
         inputs: Union[NodeSet, NodeList],
+        visited: Optional[NodeSet] = None,
     ):
         """
         Start from inputs and going reverse topological order. If any upstream node
         is in the fusion group, add all the nodes in this path to fusion group.
         """
         for arg in inputs:
+            # skip the node if already seen
+            if visited is not None:
```

Comment:
then drop this if

Reply:
Can't do that, due to the above. Actually I had non-`Optional` argument in the first version, but PT's BC linter wasn't happy saying "this is a new parameter, gotta make it optional".